### ISSUE TITLE
Is it possible to have Federated Learning on Cloud-Edge?

### ISSUE THREAD

**Issue by Martiniann** [IBM/federated-learning-lib#32]
Hi everyone,

Currently I am working on a school project about federated learning and came across your framework during exploratory analysis. My project should utilize federated learning in this manner - I have an aggregation server (let's say in a cloud). I want this server to provide model to my 2 Raspberry PIs. These two RPIs would then train the model on a local data for x epochs and provide the trained models/gradients back to the global server. On this server, the results would be federated averaged and new model would be sent to the PIs. Is such a workflow possible with your framework? If so, could you provide me a hint?

Thank you,
Best regards

**Comment by chalianwar** [comment]
Hi, thank you for looking in to IBM FL. Yes, this is one of the intended use case for this framework. You can launch Aggregator in the cloud and connect Raspberry PIs to it. You just need to update the configuration file to reflect proper IP and ports. Here are the setup steps:
https://github.com/IBM/federated-learning-lib/blob/main/setup.md

There are a number of examples with explanation for different federated learning tasks with different model types to get started with.
https://github.com/IBM/federated-learning-lib/blob/main/examples/README.md

Following has couple of Jupyter notebooks which are for non distributed version but you can easily convert them in to a distributed setup:
https://github.com/IBM/federated-learning-lib/tree/main/examples/keras_classifier
